### PR TITLE
Make it possible to configure labels and annotations for PVCs

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Enhancement / new feature

### Description

As discussed in #1914, this PR adds the possibility to specify labels and annotations for Persistent Volumes Claims. It is using the `template` section to cofnigure this.

### Checklist

- [x] Write tests
- [x] Make sure all tests pass
- [x] Update documentation
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [x] Reference relevant issue(s) and close them after merging
- [x] Update CHANGELOG.md

### CODE REVIEW

File: documentation/book/appendix_crds.adoc [hunk 1]
```diff
@@ -795,6 +797,21 @@ Used in: xref:type-EntityOperatorTemplate-{context}[`EntityOperatorTemplate`], x
...
+[id='type-PodDisruptionBudgetTemplate-{context}']
+### `PodDisruptionBudgetTemplate` schema reference
+
+Used in: xref:type-KafkaBridgeTemplate-{context}[`KafkaBridgeTemplate`], xref:type-KafkaClusterTemplate-{context}[`KafkaClusterTemplate`], xref:type-KafkaConnectTemplate-{context}[`KafkaConnectTemplate`], xref:type-KafkaMirrorMakerTemplate-{context}[`KafkaMirrorMakerTemplate`], xref:type-ZookeeperClusterTemplate-{context}[`ZookeeperClusterTemplate`]
+
+
+[options="header"]
+|====
+|Property               |Description
+|metadata        1.2+<.<|Metadata which should be applied to the `PodDistruptionBugetTemplate` resource.
```
Comment (PaulRMellor) [inline documentation/book/appendix_crds.adoc]:
```suggestion
|metadata        1.2+<.<|Metadata to apply to the `PodDistruptionBugetTemplate` resource.
```

File: documentation/book/appendix_crds.adoc [hunk 2]
```diff
@@ -795,6 +797,21 @@ Used in: xref:type-EntityOperatorTemplate-{context}[`EntityOperatorTemplate`], x
...
+|xref:type-MetadataTemplate-{context}[`MetadataTemplate`]
+|maxUnavailable  1.2+<.<|Maximum number of unavailable pods to allow voluntary Pod eviction. A Pod eviction will only be allowed when "maxUnavailable" or fewer pods are unavailable after the eviction. Setting this value to 0 will prevent all voluntary evictions and the pods will need to be evicted manually. Defaults to 1.
```
Comment (PaulRMellor) [inline documentation/book/appendix_crds.adoc]:
```suggestion
|maxUnavailable  1.2+<.<|Maximum number of unavailable pods to allow voluntary Pod eviction. A Pod eviction is allowed when the `maxUnavailable` number of pods or fewer are unavailable after the eviction. Setting this value to 0 prevents all voluntary evictions, so the pods must be evicted manually. Defaults to 1.
```

Comment (PaulRMellor) [inline documentation/book/appendix_crds.adoc]: I didn't quite get the voluntary opposed to manual here: `Setting this value to 0 prevents all voluntary evictions, so the pods must be evicted manually. Defaults to 1.`

If I substitute voluntary for automatic, it makes more sense.

**Comment by scholzj** [comment]
@PaulRMellor Thanks for your comments. I will open a separate PR for your comments because they are unrelated to this change.

**Comment by scholzj** [comment]
The docs comments are now addressed in #1962

**Pull request merged**